pinery-miso/src/main/java/ca/on/oicr/pinery/lims/miso/MisoClient.java

```diff
@@ -0,0 +1,1031 @@
+
+  // Sample queries
+  private static final String queryAllSamples = getResourceAsString("queryAllSamples.sql");
+  private static final String querySampleById = "SELECT * FROM (" + queryAllSamples + ") combined " + "WHERE id = ?";
+
+  private static final String querySampleChildIdsBySampleId = getResourceAsString("querySampleChildIdsBySampleId.sql");
+
+  // SampleType (MISO SampleClass and Library) queries
+  private static final String queryAllSampleTypes = getResourceAsString("queryAllSampleTypes.sql");
+
+  // SampleProject queries
+  private static final String queryAllSampleProjects = getResourceAsString("queryAllSampleProjects.sql");
+
+  // SampleChangeLog queries
+  private static final String queryAllSampleChangeLogs = getResourceAsString("queryAllSampleChangeLogs.sql");
+  private static final String querySampleChangeLogById = "SELECT * FROM (" + queryAllSampleChangeLogs + ") combined "
+      + "WHERE sampleId = ?";
+  
+  // Box queries
+  private static final String queryAllBoxes = getResourceAsString("queryAllBoxes.sql");
+  // @formatter:on
+
+  private final RowMapper<Instrument> instrumentMapper = new InstrumentMapper();
+  private final RowMapper<InstrumentModel> modelMapper = new InstrumentModelRowMapper();
+  private final RowMapper<Order> orderMapper = new OrderRowMapper();
+  private final RowMapper<MisoOrderSample> orderSampleMapper = new OrderSampleRowMapper();
+  private final RowMapper<User> userMapper = new UserRowMapper();
+  private final RowMapper<Run> runMapper = new RunRowMapper();
+  private final RowMapper<MisoRunPosition> runPositionMapper = new RunPositionRowMapper();
+  private final RowMapper<Sample> sampleMapper = new SampleRowMapper();
+  private final RowMapper<MisoRunSample> runSampleMapper = new RunSampleRowMapper();
+  private final RowMapper<Type> typeMapper = new TypeRowMapper();
+  private final RowMapper<SampleProject> sampleProjectMapper = new SampleProjectMapper();
+  private final RowMapper<MisoChange> changeMapper = new ChangeMapper();
+  private final RowMapper<String> idListMapper = new IdListMapper();
+
+  private JdbcTemplate template;
+
+  public MisoClient(JdbcTemplate template) {
+    this.template = template;
+  }
+
+  public MisoClient(DataSource dataSource) {
+    this.template = new JdbcTemplate(dataSource);
+  }
+
+  public JdbcTemplate getJdbcTemplate() {
+    return template;
+  }
+
+  public void setJdbcTemplate(JdbcTemplate template) {
+    this.template = template;
+  }
+
+  @Override
+  public Sample getSample(String id) {
+    validateSampleId(id);
+    List<Sample> samples = template.query(querySampleById, new Object[] { id }, sampleMapper);
+    return samples.size() == 1 ? addChildren(samples.get(0)) : null;
+  }
+
+  private void validateSampleId(String id) {
+    if (id != null && id.length() > 3) {
+      try {
+        Integer.parseInt(id.substring(3, id.length()));
+        switch (id.substring(0, 3)) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a default case to this switch. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)